addons/sale_coupon/tests/test_program_numbers.py

```diff
@@ -790,10 +787,38 @@ def test_coupon_and_coupon_discount_fixed_amount_tax_incl(self):
                 'coupon_code': 'test_10pc'
             }).process_coupon()
         order.recompute_coupon_lines()
-        #TODO fix numbers
-        self.assertEqual(order.amount_tax, 9.01)
-        self.assertEqual(order.amount_untaxed, 0.08)
-        self.assertEqual(order.amount_total, 9.09, "The promotion program should not be altered after recomputation")
+        self.assertEqual(order.amount_total, 0.0, "The promotion program should not be altered after recomputation")
+
+    def test_program_discount_different_percentages_with_tax_mapping(self):
+        # test 100% percentage discount (tax included)
+
+        p = self.env['sale.coupon.program'].create({
```

Comment:
You would not use `p` as a variable name in business code outside of test, it doesn't mean anything and brings confusion when reading code a few lines below: "What is p variable :shrug: ?"
Let's do the same in tests, they gotta be readable the same way regular code base, this is especially true since when they are read, that's because of a red runbot when writting a tests that is breaking a flow. Thus, understanding the test without too much pain is something to consider :)

Note that I am not trying to be annoying or anything here, I just read the test for final review and had to scroll up to see what was `p` variable.